### ISSUE TITLE
The Exponentiation operator causes Parsing error

### ISSUE THREAD

**Issue by dotnetCarpenter** [eslint/eslint#10481]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** v8.11.1
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
node_modules/.bin/eslint --init
? How would you like to configure ESLint? Inspect your JavaScript file(s)
? Which file(s), path(s), or glob(s) should be examined? src tests
? What format do you want your config file to be in? JavaScript
? Are you using ECMAScript 6 features? Yes
? Are you using ES6 modules? Yes
? Where will your code run? Browser
? Do you use CommonJS? Yes
? Do you use JSX? No
```

**What did you expect to happen?**
A generated config file.

**What actually happened? Please include the actual, raw output from ESLint.**

```
(/Users/nappdev/projects/opensource/panzoom2/src/models/Point.js:27:16) Parsing error: Unexpected token *
Error: (/Users/nappdev/projects/opensource/panzoom2/src/models/Point.js:27:16) Parsing error: Unexpected token *
    at getSourceCodeOfFile (/Users/nappdev/projects/opensource/panzoom2/node_modules/eslint/lib/util/source-code-util.js:38:15)
    at filenames.forEach.filename (/Users/nappdev/projects/opensource/panzoom2/node_modules/eslint/lib/util/source-code-util.js:91:28)
    at Array.forEach (<anonymous>)
    at getSourceCodeOfFiles (/Users/nappdev/projects/opensource/panzoom2/node_modules/eslint/lib/util/source-code-util.js:90:15)
    at configureRules (/Users/nappdev/projects/opensource/panzoom2/node_modules/eslint/lib/config/config-initializer.js:165:23)
    at processAnswers (/Users/nappdev/projects/opensource/panzoom2/node_modules/eslint/lib/config/config-initializer.js:288:18)
    at inquirer.prompt.then.secondAnswers (/Users/nappdev/projects/opensource/panzoom2/node_modules/eslint/lib/config/config-initializer.js:536:32)
    at <anonymous>
Determining Config: 11% [===---------------------------] 0.2s elapsed, eta 1.8s
```

The offending code is:
```js
class Point {
  constructor ({x = 0, y = 0}) {
    this.x = parseFloat(x)
    this.y = parseFloat(y)
  }

  /**
   * Substract point from this point
   * @param {Point} point
   * @returns {Point}
   */
  delta (point) {
    return new Point({
      x: this.x - point.x,
      y: this.y - point.y
    })
  }

  /**
   * Calculate the distance between 2 points
   * @param {Point} point
   * @returns {number}
   */
  distance (point) {
    const delta = this.delta(point)
    return Math.sqrt(
      delta.x ** 2 + delta.y ** 2 // <- this line throws a parsing error in eslint 4
    )
  }
}

export default Point
```

[From ES2016 specification](https://www.ecma-international.org/ecma-262/7.0/#sec-exp-operator-runtime-semantics-evaluation):

12.6.3Runtime Semantics: Evaluation#
ExponentiationExpression:UpdateExpression**ExponentiationExpression
Let left be the result of evaluating UpdateExpression.
Let leftValue be ? GetValue(left).
Let right be the result of evaluating ExponentiationExpression.
Let rightValue be ? GetValue(right).
Let base be ? ToNumber(leftValue).
Let exponent be ? ToNumber(rightValue).
Return the result of [Applying the ** operator](https://www.ecma-international.org/ecma-262/7.0/#sec-applying-the-exp-operator) with base and exponent as specified in 12.7.3.4.

**Comment by platinumazure** [comment]
I think this might be a duplicate of #10003 and fixed by #10378. This should be fixed on master and in the release candidate, and will be fixed in 5.0.0 when that comes out.

**Comment by dotnetCarpenter** [comment]
I can confirm that this issue is fixed per eslint@5.0.0-rc.0.
